kiwi/constraint.h

```diff
@@ -57,6 +58,18 @@ class Constraint
         return m_data->m_strength;
     }
 
+    bool violated() const
+    {
+        switch (m_data->m_op)
+        {
+            case OP_EQ: return !impl::nearZero(m_data->m_expression.value());
+            case OP_GE: return m_data->m_expression.value() < 0.0;
+            case OP_LE: return m_data->m_expression.value() > 0.0;
+        }
+
+        std::abort();
```

Comment:
Yep. The compiler will allow compilation without it since it's an exhaustive `enum` switch, but there's nothing stopping there being completely garbage data there, and not returning from a function that has a non-`void` return type is undefined behavior, so best to just consider it some completely unacceptable, this-should-never-happen case and abort since there's no 'sane' path forward.

Reply:
Thanks. 

I forgot, could you add a changelog entry (and bump the solver version to 1.4.0) ?

Reply:
Sure :) You might also want to update the copyright dates by the way, looks like they're 2 years out of date. Probably best as another commit though.

Reply:
I *think* I did that correctly. The two versions is confusing, but since both the bindings and the solver had API modifications I figured both needed to be bumped. Let me know if you want something a bit different.